include/contractor/contractor.hpp

```diff
@@ -846,22 +869,65 @@ class Contractor
                     continue;
                 }
                 const NodeID target = contractor_graph->GetTarget(out_edge);
+                if (node == target)
+                    continue;
+                if (target == source)
+                {
+                    EdgeWeight loop_weight = in_data.distance + out_data.distance;
+                    if (loop_weight < node_weights[node])
+                    {
+                        if (RUNSIMULATION)
+                        {
+                            // make sure to prune better, but keep inserting this loop if it should
+                            // still be the best
+                            // CAREFUL: This only works due to the independent node-setting. This
+                            // guarantees that source is not connected to another node that is
+                            // contracted
+                            node_weights[source] = loop_weight + 1;
+                            BOOST_ASSERT(stats != nullptr);
+                            stats->edges_added_count += 2;
+                            stats->original_edges_added_count +=
+                                2 * (out_data.originalEdges + in_data.originalEdges);
+                        }
+                        else
+                        {
+                            // CAREFUL: This only works due to the independent node-setting. This
+                            // guarantees that source is not connected to another node that is
+                            // contracted
+                            node_weights[source] = loop_weight + 1;
+                            node_weights[source] = loop_weight; // make sure to prune better
+                            inserted_edges.emplace_back(
+                                source, target, loop_weight,
+                                out_data.originalEdges + in_data.originalEdges, node, SHORTCUT_ARC,
+                                FORWARD_DIRECTION_ENABLED, REVERSE_DIRECTION_DISABLED);
+
+                            inserted_edges.emplace_back(
+                                target, source, loop_weight,
+                                out_data.originalEdges + in_data.originalEdges, node, SHORTCUT_ARC,
+                                FORWARD_DIRECTION_DISABLED, REVERSE_DIRECTION_ENABLED);
+                        }
+                    }
+                    continue;
+                }
                 const int path_distance = in_data.distance + out_data.distance;
```

Comment:
`loop_weight` and `path_distance` seem to always have the same value, but are declared in different scopes. Move `path_distance` up.